### ISSUE TITLE
NVDA does not read the file transfer information in Skype

### ISSUE THREAD

**Issue by nvaccessAuto** [nvaccess/nvda#3761]
**Reported by k_kolev1985 on 2014-01-08 18:51**
In newer versions of Skype (from 4.0 and above), the file transfers and their related information is displayed as chat message entries in the conversation window. I don't know about before, but now the information in those entries is exposed in an accessible for NVDA way. The problem however is the fact that the name of the file, the remaining time, the progress bar and the "Cancel" button are exposed as child objects to the main entry for the file transfer entry (object) and NVDA seams not to automatically parse those children objects. Therefore, NVDA does not announce those child objects when the system (keyboard) focus lands on the parent object for the file transfer in the chat conversation list (the chat messages list). Can't something be done within NVDA, in order to improve the experience while working with file transfers in Skype?

Steps to reproduce it:
1. Open Skype.
2. Start a conversation (chat) with someone and initiate a file transfer with the person (it doesn't matter if the file transfer is incoming or outgoing).
3. With Shift+TAB and then with the arrow keys, move the system (keyboard) focus to the current file transfer entry in the list of chat messages. NVDA will report only the file name, the name of the sender and the date and time of sending (if I'm not mistaking).
4. While the system (keyboard) focus is on the file transfer entry, if the object navigation cursor is not on the focused object, move it to it.
5. Press NVDA+NumPad2 to go in the children object hierarchy of the file transfer entry (object).
6. Then with NVDA+NumPad4 and NVDA+NumPad6 explore the objects inside. You'll find that the file name, the remaining time, the progress bar and the "Cancel" button are seperate objects in that level of the hierarchy and NVDA then reads them fine.

Expected results: When the system (keyboard) focus is placed on the file transfer entry in the chat messages list, NVDA should read not only the information for that entry specifically placed there for the screen readers, but the children objects in that entry (object) as well, in order to provide more information about the file transfer in question.

Actual results: NVDA reads only the information in that file transfer entry, placed specifically there for the screen readers.

Notes:
1. Although the progress bar is present in the children level of the hierarchy for the file transfer entry (object), it doesn't seam to provide percentage indication for the file transfer.
2. There needs to be done something in colaboration with Skype, in order to improve even more the experience with file transfers in Skype. Because now, there is even more useful information shown for an ongoing file transfer, but it is displayed in a tooltip when the mouse cursor hovers over the icon of the transfering file entry and it is not directly accessible for screen readers. Even if the user manages to point the mouse cursor on that icon, the tooltip displayed is not read by NVDA (a ticket for the unreadable tooltips in some applications like Skype is already filed by me in this tracker - see ticket #3314).
3. In this ticket I'm refering only to the file transfer entries (objects) with such behavior, but this may affect other similar objects in Skype - I just haven't noticed it yet).

My configuration:
1. Skype version: 6.11.0.102.
2. NVDA version: latest main snapshot, but the result is the same with the stable 2013.3 as well.
3. OS: Windows 7 Ultimate with SP1, 32-bit.

**Comment by Adriani90** [comment]
@k-kolev1985 is this still the case with newest skype versions?

**Comment by k-kolev1985** [comment]
@Adriani90 Well, I think it is for Skype 7, but that version is going away at some point in the near future in favor of Skype 8 and newer.

I haven't tested file transfer in Skype 8 recently, since I rarely use Skype's file transfer functionality nowadays. But if there some issues with it, I suggest for them to be filed in other tickets.

I think this ticket should be closed, since there is no sense to work on improvements for a version of a program that is probably going away at some point in the near future.

**Comment by ehollig** [comment]
I will take your suggestion @k-kolev1985 and close this issue. If there are issues with the newer version of Skype, please open a new issue. Closing
